src/GraphQL/Value/Item.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformGraphQL\GraphQL\Value;
+
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use EzSystems\EzPlatformGraphQL\GraphQL\Resolver\LocationGuesser\LocationGuesser;
+
+/**
+ * A DXP item, combination of a Content and Location.
+ */
+class Item
+{
+    /** @var \eZ\Publish\API\Repository\Values\Content\Content */
+    private $content;
+
+    /** @var \eZ\Publish\API\Repository\Values\Content\Location */
+    private $location;
+
+    /** @var \EzSystems\EzPlatformGraphQL\GraphQL\Resolver\LocationGuesser\LocationGuesser */
+    private $locationGuesser;
+
+    public function __construct(LocationGuesser $locationGuesser, ?Location $location = null, ?Content $content = null)
+    {
+        if ($location === null && $content === null) {
+            // @todo find a better exception
+            throw new \Exception('Constructing an item requires one of location or content');
+        }
```

Comment:
... or mark constructor as private/protected and force class instantiation using named constructors: 

```
public static fromContent(LocationGuesser $locationGuesser, Content $content): self;
```

```
public static fromLocation(LocationGuesser $locationGuesser, Content $location): self;
```